lab-katherine/__test__/route/route-toy.test.js

```diff
@@ -0,0 +1,163 @@
+'use strict'
+
+const server = require('../../lib/server.js')
+const superagent = require('superagent')
+
+describe('testing toy routes', function(){
+  beforeAll(done => server.listen(3000, done))
+  afterAll((done) => {
+    server.close(() => done())
+  })
+
+  //NOTE: passed and failed!
+  describe('no endpoint', () => {
+    test('return 404 if routes not registered', done => {
+      superagent.post(':3000/api/death')
+        .set('Content-Type', 'application/json')
+        .send({})
+        .end((err, res) => {
+          expect(err).not.toBeNull()
+          expect(res.status).toBe(404)
+          done()
+        })
+    })
+  })
+
+  describe('all requests to /api/toy', () => {
+
+    describe('#POST method', () => {
+      //NOTE:
+      test.only('201 should create and return a new toy, given a valid request', done => {
+        superagent.post(':3000/api/toy')
+          .type('application/json')
+          .send({
+            name: 'barney',
+            desc: 'purple dino'
+          })
+          .end(res => {
+            //can do text or body maybe
+            console.log('body', res)
+            // let mockToy = JSON.parse(res.body)
+            expect(res.status).toEqual(201)
+            console.log('after the res statut')
+            expect(res.body.name).toEqual('bernie')
+            console.log('after the body')
+            expect(res.body.desc).toEqual('purple dino')
+            console.log('after the res desc')
+            expect(res.body).toHaveProperty('_id')
+            // done()
+          })
+        done()
```

Comment:
All of these 'done()'s that are after the then/end/catch blocks are not necessary.